lib/local-cluster/lib/Cardano/Wallet/Launch/Cluster/Faucet/Serialize.hs

```diff
@@ -0,0 +1,144 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Cardano.Wallet.Launch.Cluster.Faucet.Serialize
+    ( retrieveFunds
+    , saveFunds
+    ) where
+
+import Prelude
+
+import Cardano.Address
+    ( bech32
+    , fromBech32
+    )
+import Cardano.Wallet.Launch.Cluster.Cluster
+    ( FaucetFunds (..)
+    )
+import Cardano.Wallet.Launch.Cluster.FileOf
+    ( FileOf
+    , absFilePathOf
+    )
+import Cardano.Wallet.Primitive.Types.AssetId
+    ( AssetId (..)
+    )
+import Cardano.Wallet.Primitive.Types.Coin
+    ( Coin (..)
+    )
+import Cardano.Wallet.Primitive.Types.TokenBundle
+    ( TokenBundle (..)
+    , fromFlatList
+    , toFlatList
+    )
+import Cardano.Wallet.Primitive.Types.TokenQuantity
+    ( TokenQuantity (..)
+    )
+import Data.Aeson
+    ( FromJSON (parseJSON)
+    , KeyValue ((.=))
+    , ToJSON (toJSON)
+    , Value (Number, Object, String)
+    , object
+    , withObject
+    , (.:)
+    )
+import Data.Aeson.Types
+    ( Parser
+    )
+import Data.Bifunctor
+    ( Bifunctor (..)
+    )
+import UnliftIO
+    ( SomeException
+    , catch
+    )
+
+import qualified Data.Text as T
+import qualified Data.Yaml as Y
+
+newtype FaucetFundsYaml = FaucetFundsYaml {getFaucetFunds :: FaucetFunds}
+
+instance Y.FromJSON FaucetFundsYaml where
+    parseJSON = fmap FaucetFundsYaml . faucetFundsFromValue
+
+instance Y.ToJSON FaucetFundsYaml where
+    toJSON (FaucetFundsYaml funds) = faucetFundsToValue funds
+
+faucetFundsToValue :: FaucetFunds -> Value
+faucetFundsToValue FaucetFunds{..} =
+    object
+        [ "pureAdaFunds" .= pureAdaFunds'
+        , "maryAllegraFunds" .= maryAllegraFunds'
+        , "massiveWalletFunds" .= massiveWalletFunds'
+        ]
+  where
+    pureAdaFunds' = encodeAddrCoins pureAdaFunds
+    maryAllegraFunds' =
+        map
+            (bimap bech32 encodeTokenBundle)
+            maryAllegraFunds
+    massiveWalletFunds' = encodeAddrCoins massiveWalletFunds
+    encodeAddrCoins = map (\(addr, Coin coin) -> (bech32 addr, coin))
+
+    encodeTokenBundle :: (TokenBundle, [(String, String)]) -> Value
+    encodeTokenBundle (bundle, keys) =
+        let (Coin c, assets) = toFlatList bundle
+        in  object
+                [ "coin" .= c
+                , "assets" .= (encodeAssets <$> assets)
+                , "keys" .= keys
+                ]
```

Comment:
```suggestion
    encodeTokenBundle :: (TokenBundle, [(String, String)]) -> Value
    encodeTokenBundle (bundle, keys) =
        let (Coin c, assets) = toFlatList bundle
        in  object
                [ "coin" .= c
                , "assets" .= (encodeAssets <$> assets)
                , "keys" .= toJSON $ map encodeKeyPair keys
                ]
    encodeKeyPair :: (String, String) → Value
    encodeKeyPair (x,y) = object [ "signingKey" .= x, "verificationKeyHash" .= y ]  
```

Perhaps it's a good idea to be slightly more verbose about what these fields mean when encoding? The type is not very informative. 🤔